cmd/test.go

```diff
@@ -0,0 +1,206 @@
+		if imagePath == "" {
+			return fmt.Errorf("Please supply path to image or tarball to test against")
+		}
+		if metadata != "" {
+			return fmt.Errorf("Cannot provide both image path and metadata file")
+		}
+	}
+	if len(configFiles) == 0 {
+		return fmt.Errorf("Please provide at least one test config file")
+	}
+	return nil
+}
+
+func RunTests() []*unversioned.FullResult {
+	fullResults := make([]*unversioned.FullResult, 0)
+	for _, file := range configFiles {
+		tests, err := Parse(file)
+		if err != nil {
+			logrus.Fatalf("Error parsing config file: %s", err)
+		}
+		fullResults = append(fullResults, &unversioned.FullResult{
+			FileName: file,
+			Results:  tests.RunAll(),
+		})
+	}
+	return fullResults
+}
+
+func Parse(fp string) (types.StructureTest, error) {
+	testContents, err := ioutil.ReadFile(fp)
+	if err != nil {
+		return nil, err
+	}
+
+	// We first have to unmarshal to determine the schema version, then we unmarshal again
+	// to do the full parse.
+	var unmarshal types.Unmarshaller
+	var strictUnmarshal types.Unmarshaller
+	var versionHolder types.SchemaVersion
+
+	switch {
+	case strings.HasSuffix(fp, ".json"):
+		unmarshal = json.Unmarshal
+		strictUnmarshal = json.Unmarshal
+	case strings.HasSuffix(fp, ".yaml"):
+		unmarshal = yaml.Unmarshal
+		strictUnmarshal = yaml.UnmarshalStrict
+	default:
+		return nil, errors.New("Please provide valid JSON or YAML config file")
+	}
+
+	if err := unmarshal(testContents, &versionHolder); err != nil {
+		return nil, err
+	}
+
+	version := versionHolder.SchemaVersion
+	if version == "" {
+		return nil, errors.New("Please provide JSON schema version")
+	}
+
+	var st types.StructureTest
+	if schemaVersion, ok := types.SchemaVersions[version]; ok {
+		st = schemaVersion()
+	} else {
+		return nil, errors.New("Unsupported schema version: " + version)
+	}
+
+	strictUnmarshal(testContents, st)
+
+	tests, ok := st.(types.StructureTest) //type assertion
+	if !ok {
+		return nil, errors.New("Error encountered when type casting Structure Test interface")
+	}
+	tests.SetDriverImpl(driverImpl, *args)
+	return tests, nil
+}
+
+func Run() []*unversioned.FullResult {
+	args = &drivers.DriverConfig{
+		Image:    imagePath,
+		Save:     save,
+		Metadata: metadata,
+	}
+
+	var err error
+
+	if pull {
+		if driver != drivers.Docker {
+			logrus.Fatal("Image pull not supported when not using Docker driver")
+		}
+		var repository, tag string
+		parts := strings.Split(imagePath, ":")
+		repository = parts[0]
+		if len(parts) < 2 {
+			logrus.Fatal("Please provide specific tag for image")
+		}
+		tag = parts[1]
+		client, err := docker.NewClientFromEnv()
+		if err = client.PullImage(docker.PullImageOptions{
+			Repository:   repository,
+			Tag:          tag,
+			OutputStream: os.Stdout,
+		}, docker.AuthConfiguration{}); err != nil {
+			logrus.Fatalf("Error pulling remote image %s: %s", imagePath, err.Error())
+		}
+	}
+
+	warnMessage := `WARNING: Using the host driver runs tests directly on the machine 
```

Comment:
nit: make this a package level constant.

Reply:
done